### PR TITLE
[enhancement] Registered menu groups #260

### LINKED ISSUE #260: [enhancement] Register menu groups
Must be done after implementing register menu group functionality https://github.com/openwisp/openwisp-utils/issues/189

### PR DESCRIPTION
**Pull request by niteshsinha17**
#### Changes

Registered menu groups

#### Checklist

- [x] I have read the [contributing guidelines](http://openwisp.io/docs/developer/contributing.html#how-to-commit-your-changes-properly).
- [x] I have manually tested the proposed changes.
- [x] I have written new test cases to avoid regressions. (if necessary)
- [x] I have updated the documentation. (e.g. README.rst)

closes #260

### CODE REVIEW

File: openwisp_radius/apps.py [hunk 1]
```diff
@@ -104,13 +104,13 @@ def radiusorgsettings_post_save(self, instance, **kwargs):
...
+    def regiser_menu_groups(self):
+        register_menu_group(
+            position=30,
+            config={
+                'label': _('Radius sessions'),
+                'model': f'{self.label}.RadiusAccounting',
+                'name': 'changelist',
+                'icon': 'radius-sessions',
+            },
+        )
```
Comment (nemesisdesign) [inline openwisp_radius/apps.py:116]: please register all the models of openwisp-radius in the group

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

File: openwisp_radius/apps.py (hunk 1 above)
Comment (nemesisdesign) [reply openwisp_radius/apps.py:116]: any idea why the tests are failing?

**Review by atb00ker — APPROVED** [review]
Hmm, not sure if radius-token also needs to be registered, I'll wait for Fed to confirm that, but the rest looks good! 😄

**Review by atb00ker — COMMENTED** [review]
Wait, icons are not visible!

File: requirements.txt [hunk 2]
```diff
@@ -1,13 +1,14 @@
 django>=3.0,<3.2
```
[the remark this answers is not in the document]
Comment (niteshsinha17) [reply requirements.txt:1]: Are you pointing to django version?

[the remark this answers is not in the document]
Comment (atb00ker) [reply requirements.txt:1]: Oops, no, I meant `niteshsinha17/openwisp-utils`.

[the remark this answers is not in the document]
Comment (niteshsinha17) [reply requirements.txt:1]: Yes, I have written at line 4

File: openwisp_radius/apps.py [hunk 3]
```diff
@@ -104,13 +105,61 @@ def radiusorgsettings_post_save(self, instance, **kwargs):
...
+                        'model': get_model_name(self.label, 'RadiusPostAuth'),
+                        'name': 'changelist',
+                        'icon': 'ow-radius-post-log',
+                    },
+                    7: {
+                        'label': _('Radius Token'),
+                        'model': get_model_name(self.label, 'RadiusToken'),
+                        'name': 'changelist',
+                        'icon': 'ow-radius-token',
+                    },
```
Comment (nemesisdesign) [inline openwisp_radius/apps.py:155]: we don't need to show this I believe, it's not useful to end users but only to us developers. Maybe we can show it only if DEBUG is True, what do you think @atb00ker?

File: openwisp_radius/apps.py [hunk 4]
```diff
@@ -104,13 +105,61 @@ def radiusorgsettings_post_save(self, instance, **kwargs):
...
-            {'model': f'{self.label}.RadiusAccounting', 'label': _('Radius sessions')}
-        ]
-        if not hasattr(settings, menu_setting):
-            setattr(settings, menu_setting, items)
-        else:
-            current_menu = getattr(settings, menu_setting)
-            current_menu += items
+    def regiser_menu_groups(self):
+        register_menu_group(
+            position=30,
```
Comment (nemesisdesign) [inline openwisp_radius/apps.py:110]: let's make sure the radius module comes after organization and users.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
here's how I think we should order this:

- RADIUS  (let's call it RADIUS not Freeradius)
  - Accounting Sessions (I'd try another icon)
  - Groups  (icon is good)
  - NAS  (icon is good)
  - Checks  (icon is good)
  - Replies   (icon looks good)
  - Batch User Creation  (icon is good)
  - Post Auth Log   (can you look if you can find another icon which is more suited to indicate logs?)

For accounting sessions, what about an icon like the following?

income

We need an icon that conveys the idea of a list of numbers, accounting sessions contain the information of when and how users accessed the nework, how much traffic they generated, when the session started and ended and why (if they logged out explicitly or if the session went into idle timeout), the mac address of their device and the device which they used to connect, etc. This is one of the most important pieces of information of this module.

File: openwisp_radius/apps.py (hunk 4 above)
Comment (niteshsinha17) [reply openwisp_radius/apps.py:110]: Okay, I will change it as now we are registering org and users at 40 now

File: openwisp_radius/apps.py (hunk 3 above)
Comment (atb00ker) [reply openwisp_radius/apps.py:155]: Yes, that sounds good to me! 😄

**Review by nemesisdesign — COMMENTED** [review]
Looking great :+1: 
I'll be doing more testing before merging.

**Review by nemesisdesign — DISMISSED** [review]

**Review by nemesisdesign — APPROVED** [review]

**Comment by kamissah** [comment]
Good work. Am new to this project, how can I use this project to manage my WISP business? Kindly assist

**Comment by niteshsinha17** [comment]
Thanks @kamissah To use openWISP you can start with the documentation https://openwisp.io/docs/user/quickstart.html You can also ask your doubts in the group chat https://gitter.im/openwisp/general

**Pull request merged**